File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/termsandconditions/TermsAndConditionsApiControllerTest.java
package io.zenwave360.example.clinicaltool.modules.termsandconditions;

import io.zenwave360.example.clinicaltool.modules.termsandconditions.config.ServicesInMemoryConfig;
import io.zenwave360.example.clinicaltool.modules.termsandconditions.dtos.*;
import java.math.*;
import java.time.*;
import java.util.*;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Test controller for TermsAndConditionsApiController.
 */
public class TermsAndConditionsApiControllerTest {

    private final Logger log = LoggerFactory.getLogger(getClass());

    ServicesInMemoryConfig context = new ServicesInMemoryConfig();

    TermsAndConditionsApiController controller =
            new TermsAndConditionsApiController(context.termsAndConditionsService());

    @BeforeEach
    void setUp() {
        context.reloadTestData();
    }

    @Test
    public void listTermsAndConditionsTest() {

        var response = controller.listTermsAndConditions();
        Assertions.assertEquals(200, response.getStatusCode().value());
    }

    @Test
    public void createTermsAndConditionsTest() {
        TermsAndConditionsDTO reqBody = null;
        var response = controller.createTermsAndConditions(reqBody);
        Assertions.assertEquals(201, response.getStatusCode().value());
    }

    @Test
    public void getTermsAndConditionsTest() {
        Long id = null;
        var response = controller.getTermsAndConditions(id);
        Assertions.assertEquals(200, response.getStatusCode().value());
    }

    @Test
    public void updateTermsAndConditionsTest() {
        Long id = null;
        TermsAndConditionsDTO reqBody = null;
        var response = controller.updateTermsAndConditions(id, reqBody);
        Assertions.assertEquals(200, response.getStatusCode().value());
    }

    @Test
    public void getCurrentTermsAndConditionsTest() {
        String lang = null;
        var response = controller.getCurrentTermsAndConditions(lang);
        Assertions.assertEquals(200, response.getStatusCode().value());
    }

    @Test
    public void acceptTermsAndConditionsTest() {
        AcceptedTermsAndConditionsInputDTO reqBody = null;
        var response = controller.acceptTermsAndConditions(reqBody);
        Assertions.assertEquals(200, response.getStatusCode().value());
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/termsandconditions/TermsAndConditionsApiControllerTest.kt
package io.zenwave360.example.clinicaltool.modules.termsandconditions

import io.zenwave360.example.clinicaltool.modules.termsandconditions.*
import io.zenwave360.example.clinicaltool.modules.termsandconditions.dtos.*
import io.zenwave360.example.clinicaltool.modules.termsandconditions.config.ServicesInMemoryConfig

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

import java.math.*
import java.time.*
import java.util.*

/**
 * Test controller for TermsAndConditionsApiController.
 */
class TermsAndConditionsApiControllerTest {

    private val log: Logger = LoggerFactory.getLogger(javaClass)

    private val context = ServicesInMemoryConfig()

    private val controller = TermsAndConditionsApiController( context.termsAndConditionsService() )

    @BeforeEach
    fun setUp() {
        context.reloadTestData()
    }


    @Test
    fun listTermsAndConditionsTest() {
        
        val response = controller.listTermsAndConditions()
        Assertions.assertEquals(200, response.statusCode.value())
    }

    @Test
    fun createTermsAndConditionsTest() {
        val reqBody: TermsAndConditionsDTO = TermsAndConditionsDTO(content = "aaa", lang = "aaa", contentVersion = "aaa", startDate = LocalDate.now())
        val response = controller.createTermsAndConditions(reqBody)
        Assertions.assertEquals(201, response.statusCode.value())
    }

    @Test
    fun getTermsAndConditionsTest() {
        val id: Long = 0L
        val response = controller.getTermsAndConditions(id)
        Assertions.assertEquals(200, response.statusCode.value())
    }

    @Test
    fun updateTermsAndConditionsTest() {
        val id: Long = 0L
val reqBody: TermsAndConditionsDTO = TermsAndConditionsDTO(content = "aaa", lang = "aaa", contentVersion = "aaa", startDate = LocalDate.now())
        val response = controller.updateTermsAndConditions(id, reqBody)
        Assertions.assertEquals(200, response.statusCode.value())
    }

    @Test
    fun getCurrentTermsAndConditionsTest() {
        val lang: String = ""
        val response = controller.getCurrentTermsAndConditions(lang)
        Assertions.assertEquals(200, response.statusCode.value())
    }

    @Test
    fun acceptTermsAndConditionsTest() {
        val reqBody: AcceptedTermsAndConditionsInputDTO = AcceptedTermsAndConditionsInputDTO(userId = 1L, termsAndConditionsId = 1L)
        val response = controller.acceptTermsAndConditions(reqBody)
        Assertions.assertEquals(200, response.statusCode.value())
    }


}
